packages/untp-test-suite/src/core/processTestSuite.ts

```diff
@@ -1,39 +1,93 @@
-import path from 'path';
-import { ConfigCredentials, TestSuite, TestSuiteResult } from './types/index';
+import {
+  ConfigContent,
+  ConfigCredentials,
+  IValidatedCredentials,
+  ICredentialTestResult,
+  IFinalReport,
+  TemplateName,
+  TestSuite,
+} from './types/index.js';
 import { readJsonFile, validateCredentialConfigs } from './utils/common.js';
 import { dynamicLoadingSchemaService } from './services/dynamic-loading-schemas/loadingSchema.service.js';
 import { hasErrors } from './services/json-schema/validator.service.js';
+import { templateMapper } from '../templates/mapper.js';
+import { generateFinalMessage, getTemplateName } from '../interfaces/utils/common.js';
+
+export const processCheckDataBySchema = async (credentialConfig: ConfigContent): Promise<IValidatedCredentials> => {
+  try {
+    const { type, version, dataPath } = credentialConfig;
+    const [schema, data] = await Promise.all([dynamicLoadingSchemaService(type, version), readJsonFile(dataPath)]);
+
+    const errors = hasErrors(schema, data);
+
+    return {
+      ...credentialConfig,
+      errors,
+    };
+  } catch (e) {
+    const error = e as Error;
+    throw new Error(`Failed to process check data by schema. ${error.message}`);
+  }
+};
+
+export const generateTestSuiteResultByTemplate = async (
+  testSuiteResultAjv: IValidatedCredentials[],
+): Promise<ICredentialTestResult[]> => {
+  try {
+    const credentialsTemplate = [] as ICredentialTestResult[];
+    for (const value of testSuiteResultAjv) {
+      const templateName = getTemplateName(value);
+      const message = await templateMapper(templateName, value);
+
+      credentialsTemplate.push(JSON.parse(message));
+    }
```

Comment:
The loop handles asynchronous by `for` will run sequentially. Should we use `map` to run parallel?